### PR TITLE
refactor - migrate simple components to stateless functional components [ready]

### PR DESCRIPTION
**Pull request by hannahstwinter**
This PR is one of a series to migrate some of our simple components within this library to stateless functional components.

### CODE REVIEW

File: src/DetailItem.jsx [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-    );
-  }
-}
+const DetailItem = (props) => {
+  return (
+    <li { ...props } className={ classnames('rs-detail-item', props.className) }>
+      { props.children }
+    </li>
+  );
+};
```
Comment (rogeriochaves) [inline src/DetailItem.jsx]: if you remove the brackets you can also remove the `return` and parens here and in a lot of other components, even less boilerplate

**Comment by rogeriochaves** [comment]
💯

File: src/DetailItem.jsx (hunk 1 above)
Comment (hannahstwinter) [inline src/DetailItem.jsx]: Yeah, I played with that in a few places... I feel like it reads better when the components are smaller and confined to one line or so. Though, now that I say that out loud I don't know that that's entirely true... I can nuke the return statements, the point of this is further simplifying very simple components. :+1:

**Comment by lairjr** [comment]
LGTM

**Comment by adampickeral** [comment]
:+1:

**Comment by MACSkeptic** [comment]
<3 <3 <3

**Pull request merged**